effekt/shared/src/main/scala/effekt/core/optimizer/Deadcode.scala

```diff
@@ -1,22 +1,25 @@
 package effekt
 package core
+package optimizer
 
-import effekt.core.Block.BlockLit
-import effekt.core.Pure.ValueVar
-import effekt.core.normal.*
+import effekt.core.optimizer. { Normalizer => normal }
 
-class Deadcode(entrypoints: Set[Id], definitions: Map[Id, Definition]) extends core.Tree.Rewrite {
-
-  val reachable = Reachable(entrypoints, definitions)
+class Deadcode(reachable: Map[Id, Usage]) extends core.Tree.Rewrite {
 
   override def stmt = {
     // Remove local unused definitions
     case Scope(defs, stmt) =>
-      scope(defs.collect {
+      normal.Scope(defs.collect {
         case d: Definition.Def if reachable.isDefinedAt(d.id) => rewrite(d)
         // we only keep non-pure OR reachable let bindings
         case d: Definition.Let if d.capt.nonEmpty || reachable.isDefinedAt(d.id) => rewrite(d)
       }, rewrite(stmt))
+
+    case Reset(body) =>
+      rewrite(body) match {
+        case BlockLit(tparams, cparams, vparams, List(prompt), body) if !reachable.isDefinedAt(prompt.id) => body
+        case b => Stmt.Reset(b)
+      }
```

Comment:
Not sure, but we could also move this to a separate phase. 

in favor:
+ deadcode would really only do deadcode elimination

against:
- removing an unused reset is a bit like removing a dead binding, so it does fit
- a separate phase would potentially cost us some time and conceptual overhead.